### PR TITLE
Added feature custom cookie header

### PR DESCRIPTION
**Pull request by guyeisenbach**

### CODE REVIEW

File: src/main/java/com/perimeterx/models/PXContext.java [hunk 1]
```diff
@@ -350,10 +351,13 @@ private void parseCookies(HttpServletRequest request, boolean isMobileToken) {
             this.pxdeVerified = true;
         } else {
             Cookie[] cookies = request.getCookies();
-            String cookieHeader = request.getHeader(Constants.COOKIE_HEADER_NAME);
+            String[] cookieHeaders = cookieHeaders(getPxConfiguration())
+                    .stream()
```
Comment (chen-zimmer-px) [inline src/main/java/com/perimeterx/models/PXContext.java:355]:
```suggestion
            String[] cookieHeaders = customCookieHeaders(getPxConfiguration())
                    .stream()
```

Comment (guyeisenbach) [reply src/main/java/com/perimeterx/models/PXContext.java]: The function returns all the cookie headers, including custom and "cookie".
I changed the function name to "cookieHeadersNames"

File: src/main/java/com/perimeterx/models/PXContext.java [hunk 2]
```diff
@@ -350,10 +351,13 @@ private void parseCookies(HttpServletRequest request, boolean isMobileToken) {
...
             Cookie[] cookies = request.getCookies();
-            String cookieHeader = request.getHeader(Constants.COOKIE_HEADER_NAME);
+            String[] cookieHeaders = cookieHeadersNames(getPxConfiguration())
+                    .stream()
+                    .map(request::getHeader)
+                    .toArray(String[]::new);
             this.requestCookieNames = CookieNamesExtractor.extractCookieNames(cookies);
             setVidAndPxhd(cookies);
-            tokens.addAll(headerParser.createRawCookieDataList(cookieHeader));
+            tokens.addAll(headerParser.createRawCookieDataList(cookieHeaders));
```
Comment (chen-zimmer-px) [inline src/main/java/com/perimeterx/models/PXContext.java:360]: why does it called tokens?

Comment (guyeisenbach) [reply src/main/java/com/perimeterx/models/PXContext.java:360]: idk

File: src/main/java/com/perimeterx/utils/PXCommonUtils.java [hunk 3]
```diff
@@ -45,4 +48,15 @@ public static Map<String, String> getHeadersFromRequest(HttpServletRequest reque
         }
         return headers;
     }
+
+    public static List<String> cookieHeadersNames(PXConfiguration configuration) {
+        List<String> lst = new LinkedList<>();
```
Comment (ori-gold-px) [inline src/main/java/com/perimeterx/utils/PXCommonUtils.java:53]: Why linkedlist?

**Review by ori-gold-px — APPROVED** [review]
Looks good to me!

**Comment by ori-gold-px** [comment]
Any unit tests we should add for this?

**Pull request merged**